modules/azure/nsg.go

```
@@ -0,0 +1,194 @@
+	Description              string
+	Protocol                 string
+	SourcePortRange          string
+	DestinationPortRange     string
+	SourceAddressPrefix      string
+	DestinationAddressPrefix string
+	Access                   string
+	Priority                 int32
+	Direction                string
+}
+
+func bindRuleList(dest *[]NsgRuleSummary, source network.SecurityRuleListResultIterator) {
+	for source.NotDone() {
+		v := source.Value()
+		*dest = append(*dest, convertToNsgRuleSummary(v.Name, v.SecurityRulePropertiesFormat))
+		err := source.NextWithContext(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func convertToNsgRuleSummary(name *string, rule *network.SecurityRulePropertiesFormat) NsgRuleSummary {
+	summary := NsgRuleSummary{}
+
+	if rule.Description != nil {
+		summary.Description = *rule.Description
+	} else {
+		summary.Description = ""
+	}
+
+	summary.Name = *name
+	summary.Protocol = string(rule.Protocol)
+	summary.SourcePortRange = *rule.SourcePortRange
+	summary.DestinationPortRange = *rule.DestinationPortRange
+	summary.SourceAddressPrefix = *rule.SourceAddressPrefix
+	summary.DestinationAddressPrefix = *rule.DestinationAddressPrefix
+	summary.Access = string(rule.Access)
+	summary.Priority = *rule.Priority
+	summary.Direction = string(rule.Direction)
+	return summary
+}
+
+// FindRuleByName looks for a matching rule name
+func (summarizedRules *NsgRuleSummaryList) FindRuleByName(name string) NsgRuleSummary {
+	for _, r := range summarizedRules.SummarizedRules {
+		if r.Name == name {
+			return r
+		}
+	}
+
+	return NsgRuleSummary{}
+}
+
+// AllowsDestinationPort checks to see if the rule allows a specific destination port.
+func (summarizedRule *NsgRuleSummary) AllowsDestinationPort(port string) bool {
+	if summarizedRule.DestinationPortRange == "*" {
+		return true
+	}
+
+	low, high := ParsePortRangeString(summarizedRule.DestinationPortRange)
+	portAsInt, _ := strconv.ParseInt(port, 10, 16)
```

Don't ignore the error returned here!

---

Handled and propagated errors from the parse function.